### ISSUE TITLE
Calculate Entropy Tuning Loss in SAC/SAC Discrete

### ISSUE THREAD

**Issue by xingdi-eric-yuan** [p-christ/Deep-Reinforcement-Learning-Algorithms-with-PyTorch#65]
Hi all, 

I might have misunderstood, but shouldn't one use `self.alpha` rather than `self.log_alpha` [HERE](https://github.com/p-christ/Deep-Reinforcement-Learning-Algorithms-with-PyTorch/blob/6297608b85/agents/actor_critic_agents/SAC.py#L184)?

Thanks.

**Comment by Howuhh** [comment]
Same question! Moreover, in the [original SAC](https://github.com/rail-berkeley/softlearning/blob/12fce21a9118aec43f70180db34b2da0ef3bdf85/softlearning/algorithms/sac.py#L254) implementation, the authors  use alpha not log alpha

```python
with tf.GradientTape() as tape:
    alpha_losses = -1.0 * (
        self._alpha * tf.stop_gradient(log_pis + self._target_entropy))
    # NOTE(hartikainen): It's important that we take the average here,
    # otherwise we end up effectively having `batch_size` times too
    # large learning rate.
    alpha_loss = tf.nn.compute_average_loss(alpha_losses)
```

**Comment by Howuhh** [comment]
@xingdi-eric-yuan I think I found an explanation from the author of the other implementation:
https://github.com/ku2482/sac-discrete.pytorch/issues/3#issuecomment-561274577

**Comment by xingdi-eric-yuan** [comment]
Thanks, @Howuhh ! I recently tried comparing between using alpha vs log of alpha in that loss (in a discrete SAC setting), I can confirm that there's nothing noticeable between the agents' performance. Although I don't observe a clear advantage using log alpha (in terms of performance), at least it does not hurt.
